app/internal/page/donor_address_test.go

```diff
@@ -0,0 +1,505 @@
+package page
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/localize"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockAddressClient struct {
+	mock.Mock
+}
+
+func (m *mockAddressClient) LookupPostcode(postcode string) ([]Address, error) {
+	args := m.Called(postcode)
+	return args.Get(0).([]Address), args.Error(1)
+}
+
+func TestGetDonorAddress(t *testing.T) {
+	w := httptest.NewRecorder()
+	localizer := localize.Localizer{}
+
+	template := &mockTemplate{}
+	template.
+		On("Func", w, &donorAddressData{
```

Comment:
Specifically since `mockTemplate` there is a method called `Func`, which is the thing being mocked. We then pass just `template.Func` to the handler because I wrote the template package to provide functions (as they were only going to be used one way)